benchmarks/benchmarks/bench_lib.py

```diff
@@ -13,13 +13,29 @@ class Pad(Benchmark):
 
     param_names = ["shape", "pad_width", "mode"]
     params = [
-        [(1000,), (10, 100), (10, 10, 10)],
-        [1, 3, (0, 5)],
-        ["constant", "edge", "linear_ramp", "mean", "reflect", "wrap"],
+        [(1000,),
+         (10, 100),
+         (10, 10, 10),
+         # 50 * 512 * 512 = 13 million points = 46 MB. should be a good
+         # out of cache describing a typical usecase
+         (50, 512, 512)],
+        [1,
+         3,
+         (0, 5)],
+        ["constant",
+         "edge", "linear_ramp",
+         # mean/median/minimum/maximum all use the same code path
+         "mean",
+         # reflect/symmetric share alot of the code path
+         "reflect",
+         "wrap"],
     ]
 
     def setup(self, shape, pad_width, mode):
         self.array = np.empty(shape)
+        # Make sure to fill the array to make the OS page fault
+        # in the setup phase and not the timed phase
+        self.array.fill(1)
```

Comment:
Could you explain why this is necessary? Or point me to an explanation?

Reply:
This is important for any allocation larger than basically 4kb. It is really good to keep in so people can change `shape` and see how the benchmark changes.

I found this issue in this comment https://github.com/numpy/numpy/issues/7544#issuecomment-418806035
And opened a formal issue here:
https://github.com/numpy/numpy/issues/11919

You can see the difference in performance when coopying to a preallocated buffer, vs one that was created with `empty`. Page faulting slows down the copy by half. It is really unlikely that somebody will be asking you to pad a `empty` or `zero` array.

And the PR to update some benchmarks in numpy:
https://github.com/numpy/numpy/pull/11959

That PR pointed me to the `full` function, which is cleaner IMO.

Reply:
Thanks!